app/views/recipes/show.html.erb

```diff
@@ -15,8 +15,10 @@
       <div class="description">
         <p><%= @recipe.description %></p>
 
-        <% if @recipe.user === current_user || current_user.admin? %>
-          <%= button_to "Edit", edit_recipe_path(@recipe), class: 'ui button', method: :get %>
+        <% if !current_user.nil? %>
```

Comment:
Could you move this conditional onto the line below it with the other checks please? Alternatively, since that's becoming a bit verbose, you could also add a helper method for something like `can_edit_recipe?(recipe)` and put it in there, but that's a bit much for this change.

Could we also use `present?` here instead of nil please, that way we can get rid of the negation.